vnext/Microsoft.ReactNative/Fabric/Composition/CompositionViewComponentView.cpp

```diff
@@ -1402,6 +1391,84 @@ bool CompositionViewComponentView::ScrollWheel(facebook::react::Point pt, int32_
   return false;
 }
 
+void CompositionViewComponentView::onKeyDown(
+    const winrt::Microsoft::ReactNative::Composition::Input::KeyboardSource &source,
+    const winrt::Microsoft::ReactNative::Composition::Input::KeyRoutedEventArgs &args) noexcept {
+  auto eventCode = CodeFromVirtualKey(args.Key());
+  bool fShift = source.GetKeyState(winrt::Windows::System::VirtualKey::Shift) !=
+      winrt::Windows::UI::Core::CoreVirtualKeyStates::None;
+  bool fAlt = source.GetKeyState(winrt::Windows::System::VirtualKey::Menu) !=
+      winrt::Windows::UI::Core::CoreVirtualKeyStates::None;
+  bool fCtrl = source.GetKeyState(winrt::Windows::System::VirtualKey::Control) !=
+      winrt::Windows::UI::Core::CoreVirtualKeyStates::None;
+  bool fMeta = (source.GetKeyState(winrt::Windows::System::VirtualKey::LeftWindows) !=
+                winrt::Windows::UI::Core::CoreVirtualKeyStates::None) ||
+      (source.GetKeyState(winrt::Windows::System::VirtualKey::RightWindows) !=
+       winrt::Windows::UI::Core::CoreVirtualKeyStates::None);
+
+  if (args.OriginalSource() == tag()) {
+    facebook::react::KeyboardEvent event;
+    event.shiftKey = fShift;
+    event.ctrlKey = fCtrl;
+    event.altKey = fAlt;
+    event.metaKey = fMeta;
+
+    event.key = FromVirtualKey(args.Key(), event.shiftKey, !!(GetKeyState(VK_CAPITAL) & 1));
+    event.code = eventCode;
+    m_eventEmitter->onKeyDown(event);
+  }
+
+  for (const auto &handledKey : m_props->keyDownEvents) {
+    if (handledKey.code == eventCode && handledKey.altKey == fAlt && handledKey.ctrlKey == fCtrl &&
+        handledKey.metaKey == fMeta && handledKey.shiftKey == fShift &&
+        handledKey.eventPhase == facebook::react::HandledKeyEvent ::EventPhase::Bubbling) {
+      // Stop bubbling event if marked as handled from JS
+      args.Handled(true);
+    }
+  }
+
+  Super::onKeyDown(source, args);
+}
+
+void CompositionViewComponentView::onKeyUp(
+    const winrt::Microsoft::ReactNative::Composition::Input::KeyboardSource &source,
+    const winrt::Microsoft::ReactNative::Composition::Input::KeyRoutedEventArgs &args) noexcept {
+  auto eventCode = CodeFromVirtualKey(args.Key());
+  bool fShift = source.GetKeyState(winrt::Windows::System::VirtualKey::LeftShift) !=
```

Comment:
Is there a reason you're only checking the left side here? Ex, shouldn't you check `winrt::Windows::System::VirtualKey::Shift`?

Reply:
Fixed